v2/tools/generator/internal/astmodel/type_visitor.go

```diff
@@ -155,7 +155,11 @@ func MakeIdentityVisitOfObjectType(makeCtx MakePerPropertyContext) func(this *Ty
 			if err != nil {
 				errs = append(errs, err)
 			} else {
-				newProps = append(newProps, prop.WithType(p))
+				// only replace property if the type was changed;
+				// this allows short-circuiting below
+				if !TypeEquals(p, prop.propertyType) {
```

Comment:
Only checking `propertyType` runs the risk of discarding required changes to other properties; suggest using `PropertyDefinition.Equals()` instead to check for changes.

Reply:
Only the type of the property is changed by the visitor?

Reply:
Apologies, I didn't read the code above - was sure that other characteristics of a property could be modified by the visitor. As you were, all good.